src/IO/ITKImageInputFileFormat.cxx

```diff
@@ -140,76 +120,261 @@ read_from_file(const std::string& filename) const
+  for (it.GoToBegin(); !it.IsAtEnd(); ++it, ++stir_iter)
+  {
+    *stir_iter = static_cast<float>(it.Get());
+  }
+}
+
+/* Copy a multi ITK Image to a multi STIR Image.
+   This method expects that itk_image is already oriented to be consistent with
+   STIR x, y, z axes.
+*/
+void copy_ITK_data_to_STIR_image(const ITKImageMulti::Pointer itk_image,
+                                 STIRImageMulti* stir_image_ptr)
+{
+  STIRImageMulti::full_iterator stir_iter = stir_image_ptr->begin_all();
+  typedef itk::ImageRegionConstIterator<ITKImageMulti> IteratorType;
+  IteratorType it (itk_image, itk_image->GetLargestPossibleRegion() );
+  for ( it.GoToBegin(); !it.IsAtEnd(); ++it, ++stir_iter) {
+    // Need to do this because ITK VectorImage pixels are VariableLengthVectors
+    typename ITKImageMulti::PointType itk_coord;
+    itk_coord[0] = it.Get()[0];
+    itk_coord[1] = it.Get()[1];
+    itk_coord[2] = it.Get()[2];
+    *stir_iter = ITK_coordinates_to_STIR_physical_coordinates<ITKImageMulti, STIRImageMulti>
+      (itk_coord, *stir_image_ptr, true);
+  }
+}
+
+
+template<typename ITKImageType>
+typename ITKImageType::Pointer
+orient_ITK_image(const typename ITKImageType::Pointer itk_image_orig,
+                 const shared_ptr<ExamInfo> exam_info_sptr)
+{
+  typedef itk::OrientImageFilter<ITKImageType,ITKImageType> OrienterType;
+  typename OrienterType::Pointer orienter = OrienterType::New();
+  orienter->UseImageDirectionOn();
+  orienter->SetInput(itk_image_orig);
+
+  // We need the origin to be in the minimum x, y, z corner. This
+  // depends on the patient position
+  switch (exam_info_sptr->patient_position.get_position()) {
+  case PatientPosition::unknown_position:
+    // If unknown, assume HFS
+  case PatientPosition::HFS:
+    // HFS means currently in LPI
+    // So origin is in RAS direction
+    orienter
+      ->SetDesiredCoordinateOrientation(itk::SpatialOrientation::ITK_COORDINATE_ORIENTATION_RAS);
+    break;
+
+  case PatientPosition::HFP:
+    // HFP means currently in RAI
+    // So origin is in LPS direction
+    orienter
+      ->SetDesiredCoordinateOrientation(itk::SpatialOrientation::ITK_COORDINATE_ORIENTATION_LPS);
+    break;
+
+  case PatientPosition::FFS:
+    // FFS means currently in RPS
+    // So origin is in LAI direction
+    orienter
+      ->SetDesiredCoordinateOrientation(itk::SpatialOrientation::ITK_COORDINATE_ORIENTATION_LAI);
+    break;
+
+  case PatientPosition::FFP:
+    // FFP means currently in LAS
+    // So origin is in RPI direction
+    orienter
+      ->SetDesiredCoordinateOrientation(itk::SpatialOrientation::ITK_COORDINATE_ORIENTATION_RPI);
+    break;
+
+  default:
+    throw std::runtime_error("Unsupported patient position, not sure how to read.");
+  }
+
+  orienter->Update();
+  return orienter->GetOutput();
+}
+
+/* Convert an ITK image into an internal STIR one. */
+template<typename ITKImageType, typename STIRImageType, typename STIRConcreteType>
```

Comment:
Do we need to template this in terms of `STIRImageType`? Couldn't we just construct a `STIRConcreteType` and return it?

Reply:
Sure, but why can't we do this in the 3D case:
```
return convert_ITK_to_STIR
        <ITKImageSingle, STIRImageSingleConcrete>
        (itk_image);
```
and this in the 4D case:
```
return convert_ITK_to_STIR<ITKImageMulti, STIRImageMulti>
        (itk_image);
```

Reply:
Just did a PR to this branch. Feel free to ignore if you think there's something wrong with it, but it passes the ctests

Reply:
Oh right! Sorry, my brain wasn't working. Yes that makes sense - my solution is unnecessarily similar to the code I refactored, that is better.